### PR TITLE
fixed certificate authentication request

### PR DESCRIPTION
**Pull request by hasuniea**

### CODE REVIEW

**Comment by boxidilshan** [comment]
Make MaxTotalConnections etc. constants.

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/BSTValidator.java [hunk 1]
```diff
@@ -91,6 +92,10 @@ public Credential validate(Credential credential, RequestData requestData) throw
             String msg = "Authentication Failure occurred due to binary security token.";
             log.error(msg, e);
             throw new WSSecurityException(msg, e);
+        } catch (OAuthTokenValidationException e) {
+            String msg = "Failed to authenticate the incoming request due to oauth token validation error.";
```
Comment (prabathabey) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/BSTValidator.java]: Avoid "log and throw" as much as possible.

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/WindowsAPIUtils.java [hunk 2]
```diff
@@ -202,6 +204,19 @@ public static OAuth2TokenValidationService getOAuth2TokenValidationService() {
...
     }
 
+    public static AuthenticatorConfig getBSTAuthenticatorConfig() {
+        PrivilegedCarbonContext ctx = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        AuthenticatorConfigService authenticatorConfigService =
+                (AuthenticatorConfigService) ctx.getOSGiService(AuthenticatorConfigService.class, null);
+        AuthenticatorConfig authenticatorConfig = authenticatorConfigService.getAuthenticatorConfig("BST");
+        if (authenticatorConfig == null) {
+            String msg = "BST authenticatorConfig has not initialized.";
+            log.error(msg);
```
Comment (prabathabey) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/WindowsAPIUtils.java]: Avoid "log and throw" as much as possible.

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/BSTValidator.java (hunk 1 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/BSTValidator.java]: fixed issue https://github.com/wso2/product-mdm/pull/156

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/WindowsAPIUtils.java (hunk 2 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/WindowsAPIUtils.java]: fixed it from https://github.com/wso2/product-mdm/pull/156

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/authenticator/OAuthTokenValidationStubFactory.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+            if (maxConnectionsPerHostParam == null || maxConnectionsPerHostParam.isEmpty()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("MaxConnectionsPerHost parameter is not explicitly defined. Therefore, the default, " +
+                            "which is 2, will be used");
+                }
+            } else {
+                connectionManager.setDefaultMaxPerRoute(Integer.parseInt(maxConnectionsPerHostParam));
+            }
+
+            String maxTotalConnectionsParam = properties.getProperty("MaxTotalConnections");
```
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/authenticator/OAuthTokenValidationStubFactory.java]: use constants to get MaxTotalConnections from https://github.com/wso2/product-mdm/pull/156

**Pull request merged**